### PR TITLE
community/megatools: new aport

### PR DESCRIPTION
**Pull request by aGaTHoS**
https://megatools.megous.com
Megatools is a collection of programs for accessing Mega.nz service from a command line of your desktop or server.

### CODE REVIEW

**Comment by jirutka** [comment]
New packages must go to the testing repository.

**Comment by aGaTHoS** [comment]
so I have to move and commit again?

2017-11-04 3:28 GMT+01:00 Jakub Jirutka <notifications@github.com>:

**Comment by TBK** [comment]
If you want to build it with LibreSSL instead of OpenSSL then use this patch - [libressl.patch.zip](https://github.com/alpinelinux/aports/files/1443661/libressl.patch.zip):

```
--- a/lib/mega.c
+++ b/lib/mega.c
@@ -803,7 +803,7 @@
     return FALSE;
   }
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
   RSA_get0_key(key, &k->m, &k->e, &k->d);
   RSA_get0_factors(key, &k->q, &k->p);
   RSA_get0_crt_params(key, NULL, NULL, &k->u);
```
https://github.com/megous/megatools/search?utf8=%E2%9C%93&q=LIBRESSL_VERSION_NUMBER&type=

You also need to add this line `options="!check"` to disable abuild trying to run check.

**Comment by aGaTHoS** [comment]
I'm gonna apply it as I already prefer and use libressl.

2017-11-05 1:10 GMT+01:00 TBK <notifications@github.com>:

File: testing/megatools/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+license="GPL2"
+depends="libssl1.0"
+makedepends="curl-dev glib-dev libressl-dev gmp-dev asciidoc"
+install=""
+options="!check"
+subpackages="$pkgname-doc"
+source="https://megatools.megous.com/builds/megatools-1.9.98.tar.gz libressl.patch"
+builddir="$srcdir/$pkgname-$pkgver"
+
+prepare() {
```
Comment (TBK) [inline testing/megatools/APKBUILD]: You do not need the prepare function.

File: testing/megatools/APKBUILD [hunk 2]
```diff
@@ -8,12 +8,17 @@ url="https://megatools.megous.com"
 arch="all"
 license="GPL2"
 depends="libssl1.0"
```
Comment (ncopa) [inline testing/megatools/APKBUILD]: libssl runtime dependency should be autodetected by abuild so `depends` should be empty

**Comment by ncopa** [comment]
I cleaned it up and merged the 3 commits. Thanks!

**Pull request closed without merging**